WWST-2228: Add DTH for ezex smart sub meter(korea market)

FenMei007S opened the pull request:
@tpmanley plz review it

FenMei007S committed:
WWST-2228: Add DTH for ezex smart sub meter(korea market)

varzac commented on devicetypes/smartthings/ezex-smart-sub-meter.src/ezex-smart-sub-meter.groovy:
```diff
@@ -0,0 +1,116 @@
+        def descMap = zigbee.parseDescriptionAsMap(description)
+        log.debug "Desc Map: $descMap"
+          
+        if (descMap.cluster == "0702" && descMap.attrId == "0000") {
```
I have the same pair of comments as I had in https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3975

varzac commented on devicetypes/smartthings/ezex-smart-sub-meter.src/ezex-smart-sub-meter.groovy:
```diff
@@ -0,0 +1,116 @@
+ *
+ */
+metadata {
+    definition (name: "eZEX Smart Submeter", namespace: "smartthings", author: "SmartThings") {
```
This and https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3975 seem very similar.  What is the need for 2 different DTHs

FenMei007S replied:
#3975  have capibility switch, bu this one not

varzac replied:
Got it, thanks.

varzac commented:
The same comments in #3975 apply here as well.

greens commented on devicetypes/smartthings/ezex-smart-sub-meter.src/ezex-smart-sub-meter.groovy:
```diff
@@ -0,0 +1,116 @@
+ *
+ */
+metadata {
+    definition (name: "eZEX Smart Submeter", namespace: "smartthings", author: "SmartThings") {
```
The filename and DTH name field must match, spacing included. You could change the file, folder, and DTH name to `ezex smart submeter` but I would again prefer a more generic approach. Perhaps `zigbee-power-meter`?

greens commented on devicetypes/smartthings/ezex-smart-sub-meter.src/ezex-smart-sub-meter.groovy:
```diff
@@ -0,0 +1,116 @@
+
+def configure() {
+    // this device will send instantaneous demand and current summation delivered every 1 minute
+    sendEvent(name: "checkInterval", value: 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
```
Device Health pings the device every `checkInterval-1` minutes. This would have it pinging the device every minute. This is too short. Even if the device _does_ report every minute, the `checkInterval` value should be the reporting interval * 2 + a couple of minutes of fuzzing, but honestly I think you don't need to do anything less than 12 minutes.

FenMei007S commented:
@greens @varzac ,updated, changed name to zigbee power meter, and deviceJoinName to eZEX Smart Submeter, but zigbee power meter display in app. Is there is any bug refering it?

greens commented on devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy:
```diff
@@ -0,0 +1,109 @@
+        capability "Energy Meter"
+        capability "Power Meter"
+        capability "Refresh"
+        capability "Polling"
```
"Polling" capability is deprecated.

Add the "Health Check" capability.

greens commented:
> @greens @varzac ,updated, changed name to zigbee power meter, and deviceJoinName to eZEX Smart Submeter, but zigbee power meter display in app. Is there is any bug refering it?

@FenMei007S not sure why this would be happening. Never seen it before. If there's only one matching fingerprint and the device is matching, then the join name should be correct.

FenMei007S commented:
@greens @varzac Plz review

FenMei007S committed:
Update messages.properties

FenMei007S committed:
Update messages.properties

dkirker commented on devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy:
```diff
@@ -0,0 +1,109 @@
+        capability "Refresh"
+        capability "Health Check"
+        capability "Sensor"
+
```
So, configure() actually won't be called without capability "Configuration". Unless something has changed in Data Management over the last couple of years.

dkirker replied:
From what I can see, this is still true.

greens replied:
god dammit

greens replied:
@FenMei007S please open a new PR to add this capability.

dkirker commented on devicetypes/smartthings/zigbee-power-meter.src/zigbee-power-meter.groovy:
```diff
@@ -0,0 +1,109 @@
+                        map.name = "power"
+                        map.value = zigbee.convertHexToInt(it.value)/1000
+                }
+                if (it.clusterInt == 0x0702 && it.attrInt == 0x0000) {
```
We should probably also do here what I recommended in #3975 https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/3975/files#r258311368

dkirker reviewed (commented):
Last I had known, configure() wasn't called if the "Configuration" capability wasn't declared with the other capabilities. Maybe this has changed, but please do verify it.
